### ISSUE TITLE
clay: %u product-type is unusable

### ISSUE THREAD

**Issue by joemfb** [urbit/urbit#1903]
It produces `~` if a case exists, and blocks if you ask for a non-existent case, while accepting but ignoring the path:

```
> .^(* %cu /==/(scot %da now))
0
> .^(* %cu /==/(scot %da now)/foo)
0
> .^(* %cu /==/(scot %da now)/foo/bar)
0
> .^(* %cu /==/(scot %da now)/foo/bar/baz)
0
```

The product type is not useful, as it's not a constant:

```
.^(~ %cu /==/(scot %da now))
%nets-failed
%nets-need
$~
%nets-have
@n
```

It seems like this would be much more useful if it checked the path.

**Comment by philipcmonk** [comment]
Formally, the idea is that every path exists (indeed, it exists as both a "file" and a "directory") so long as the case exists.  Then, you can use .^(arch %cy %/foo) to get an +arch which will tell you if the path has any contents and/or children.

I could see an argument of using %u for checking whether a path has content, but I don't know the natural answer to what if it's a "directory" but not a "file".

**Comment by joemfb** [comment]
Ah, I didn't realize this. The `%y` care already covers this use-case. `%u` doesn't seem particularly useful, as the only information it can convey is whether it blocked or not (and, at least in an app, you won't be able to check!). In any case, this is not an issue.

**Comment by joemfb** [comment]
Revising that, the product of `%u` remains an issue.

**Comment by joemfb** [comment]
#1534 is somewhat relevant.

**Comment by philipcmonk** [comment]
Ah, yeah, it shouldn't block when the case doesn't exist.  I sort of feel like the point of `%u` is to check whether the scry will block, but of course we can't produce that answer in a referentially-transparent way.

Maybe it would be better to support `.^(? %u %/test-case)` to check whether `test-case` exists right now (and crash if the formal case is not equal to now).

**Comment by ohAitch** [comment]
What about `?`, which is `|` if the desk doesn't exist and the case is in the past(or otherwise prior to `/==0`), `&` if it already does/did, and blocks until the referenced time otherwise? I can see the argument for "reject non-time case", while "label or number always returns `&` resolving the build whenever relevant" is coherent it may not be exactly useful.

If it's not doing `?=(^ file:.^(arch cy+...))` it should probably crash on non-`/` spur rather than ignoring it though.

**Comment by philipcmonk** [comment]
fixed in #2119
